ecommerce/models.py

```diff
@@ -154,3 +156,97 @@ def save(self, *args, **kwargs):
     def __str__(self):
         """Description for CoursePrice"""
         return "CoursePrice for {}, price={}, is_valid={}".format(self.course_run, self.price, self.is_valid)
+
+
+class Coupon(Model):
+    """
+    Model for a coupon. This stores the discount for the coupon, how many people can use it, and how many times
+    a person can use it.
+
+    When a coupon is redeemed by a purchaser the counter on this object is decremented
+    and a UserCoupon object is created for that particular purchaser.
+    """
+    PERCENT_DISCOUNT = 'percent-discount'
+    FIXED_DISCOUNT = 'fixed-discount'
+
+    AMOUNT_TYPES = [PERCENT_DISCOUNT, FIXED_DISCOUNT]
+
+    COURSE_RUN = 'course-run'
+    COURSE = 'course'
+    PROGRAM = 'program'
+    PRODUCT_TYPES = [PROGRAM, COURSE, COURSE_RUN]
+
+    # The coupon code used for redemption by the purchaser in the user interface.
+    # If blank, the purchaser may not redeem this coupon through the user interface,
+    # though it may be redeemed in their name by an administrator.
+    coupon_code = TextField(null=True, blank=True)
+    product_type = CharField(
+        choices=[(_type, _type) for _type in PRODUCT_TYPES],
+        max_length=30,
+    )
+    product_id = IntegerField(null=False)
```

Comment:
The primary key of either `Course`, `CourseRun`, or `Program` depending on what `product_type` is